apps/condo/domains/marketplace/access/Invoice.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * Generated by `createschema marketplace.Invoice 'number:Integer; property:Relationship:Property:PROTECT; unitType:Text; unitName:Text; accountNumber:Text; toPay:Decimal; items:Json; contact?:Relationship:Contact:SET_NULL; client?:Relationship:User:SET_NULL; clientName?:Text; clientPhone?:Text; clientEmail?:Text'`
+ */
+
+const { get } = require('lodash')
+
+const { throwAuthenticationError } = require('@open-condo/keystone/apolloErrorFormatter')
+const { getById, find } = require('@open-condo/keystone/schema')
+
+const { INVOICE_STATUS_PUBLISHED } = require('@condo/domains/marketplace/constants')
+const {
+    queryOrganizationEmployeeFor,
+    queryOrganizationEmployeeFromRelatedOrganizationFor,
+    checkPermissionInUserOrganizationOrRelatedOrganization,
+} = require('@condo/domains/organization/utils/accessSchema')
+const { RESIDENT } = require('@condo/domains/user/constants/common')
+
+async function canReadInvoices ({ authentication: { item: user } }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+
+    if (user.isAdmin || user.isSupport) return {}
+
+    if (user.type === RESIDENT) {
+        const residents = await find('Resident', { deletedAt: null, user: { id: user.id } })
+
+        if (residents.length === 0) {
+            return false
+        }
+
```

Comment:
Please log it if the length is more than 50! We need to limit residents and service consumers for one user

Reply:
Please check last incident report

Reply:
Write log within access checking? Looks like this place is called too often to log something.

Reply:
ok-ok, logs were added :)